src/lib/compute_text_alternative_test.ts

```diff
@@ -41,4 +41,19 @@ describe('The computeTextAlternative function', () => {
     const elem = document.getElementById('foo');
     expect(computeTextAlternative(elem!)).toBe('Hello world');
   });
+
+  it('prefers input value to aria-label for embedded controls', () => {
+    render(
+      html`
+        <button id="foo">
```

Comment:
this is a bit weird because buttons have presentational children, so technically the input looses it's semantics in this case.

can you switch to a div with role="link" instead ?